src/sentry/workflow_engine/migrations/0061_backfill_metric_alert_resolution_action_filters.py

```diff
@@ -0,0 +1,90 @@
+# Generated by Django 5.1.7 on 2025-05-08 21:27
+from enum import Enum
+
+from django.apps.registry import Apps
+from django.db import migrations, router, transaction
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapper
+
+
+class AlertRuleStatus(Enum):
+    PENDING = 0
+    SNAPSHOT = 4
+    DISABLED = 5
+    NOT_ENOUGH_DATA = 6
+
+
+def backfill_resolution_action_filters(apps: Apps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    AlertRule = apps.get_model("sentry", "AlertRule")
+    Organization = apps.get_model("sentry", "Organization")
+
+    AlertRuleWorkflow = apps.get_model("workflow_engine", "AlertRuleWorkflow")
+    DataCondition = apps.get_model("workflow_engine", "DataCondition")
+    DataConditionGroup = apps.get_model("workflow_engine", "DataConditionGroup")
+
+    for organization in RangeQuerySetWrapper(Organization.objects.all()):
+        organization_id = organization.id
+        alert_rule_ids = AlertRule.objects_with_snapshots.filter(
+            organization_id=organization_id, status=AlertRuleStatus.PENDING.value
+        ).values_list("id", flat=True)
+
+        workflow_ids = AlertRuleWorkflow.objects.filter(
+            alert_rule_id__in=alert_rule_ids
+        ).values_list("workflow__id", flat=True)
```

Comment:
nit: You can also just use `workflow_id` here to avoid the join. Not a huge deal, jfyi